src/common/settings.ts

```diff
@@ -127,6 +149,19 @@ function getGlobalValue<T>(config: WorkspaceConfiguration, key: string, defaultV
   return inspect?.globalValue ?? inspect?.defaultValue ?? defaultValue;
 }
 
+function getOptionalGlobalValue<T>(config: WorkspaceConfiguration, key: string): T | undefined {
```

Comment:
Nit: I prefer to use `null` instead of `undefined` for "absent" values because it is explicit that the value is not set whereas a value can be `undefined` if it is not set or not initializsed.

Reply:
`config.inspect`, the method from the VS Code API that we use to fetch configuration, uses `undefined` to represent absent values, which is why I think it makes more sense to also return `undefined` here.

![Screenshot 2024-04-22 at 2 37 36 PM](https://github.com/astral-sh/ruff-vscode/assets/19577865/6a32d30f-d98a-4862-b109-31173821008b)